clients/python/src/model_registry/_client.py

```diff
@@ -950,3 +950,29 @@ def exp_run_logs(options: ListOptions) -> list[ExperimentRunArtifact]:
             return None
 
         return Pager[ExperimentRunArtifact](exp_run_logs)
+
+    def get_artifacts(
+        self,
+        filter_query: str | None = None,
+        artifact_type: str | None = None,
+    ) -> Pager[Artifact]:
+        """Get artifacts with filtering capabilities.
+
+        Args:
+            filter_query: A SQL-like query string to filter the list of entities.
+            artifact_type: Specifies the artifact type for listing artifacts.
+
+        Returns:
+            Iterable pager for artifacts.
+        """
+
+        def artifacts_list(options: ListOptions) -> list[Artifact]:
+            return self.async_runner(
+                self._api.get_artifacts(
+                    filter_query=filter_query,
+                    artifact_type=artifact_type,
+                    options=options,
+                )
+            )
+
+        return Pager[Artifact](artifacts_list)
```

Comment:
I'm thinking we don't actually want to add `get_artifacts` to our `ModelRegistry` client like this, but I added it in this draft to try this out and illustrate what I'm trying to test.

Maybe a similar function to this should be defined in our test utilities?